### ISSUE TITLE
Traceback on run INSTANCE --debug-shell

### ISSUE THREAD

**Issue by rusty-snake** [igo95862/bubblejail#69]
### Output of `bubblejail --version`

0.8.0

### Your distro name and version

Fedora 38

### Description

```console 
$ bubblejail run BubblejailTesting --debug-shell
usage: bubblejail-helper [-h] --helper-socket HELPER_SOCKET [--shell] [--ready-fd READY_FD] ...
bubblejail-helper: error: unrecognized arguments: --debug-shell
Traceback (most recent call last):
  File "/usr/bin/bubblejail", line 32, in <module>
    bubblejail_main()
  File "/usr/lib64/bubblejail/python_packages/bubblejail/bubblejail_cli.py", line 238, in bubblejail_main
    func(**args_dict)
  File "/usr/lib64/bubblejail/python_packages/bubblejail/bubblejail_cli.py", line 105, in run_bjail
    async_run(
  File "/usr/lib64/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/usr/lib64/bubblejail/python_packages/bubblejail/bubblejail_instance.py", line 252, in async_run_init
    raise BubblewrapRunError((
bubblejail.exceptions.BubblewrapRunError: Bubblewrap failed. Try running bubblejail in terminal to see the exact error.
```

**Comment by igo95862** [comment]
Anything after instance name will be passed to the instance as commands.

@rusty-snake do you have any other issues?

**Comment by rusty-snake** [comment]
To be clear, the issue here is that the users sees an Traceback instead of a helpful error.

---

> do you have any other issues?

Most confusing was the behaviour of `run foo bar` when you have a `run foo baz` running in an other terminal. Because the command exists with code 0 but you do not see anything in your terminal.

Also the README does not mention that bubblejail does not work with suid bubblewrap.

**Comment by igo95862** [comment]
> To be clear, the issue here is that the users sees an Traceback instead of a helpful error.

a3e8a8cf02469fb83135c6675d214e61d67cd265 I made sure that if you pass some `--option` without the any non option argument it is not treated as a file name to run. Maybe someone has an executable that start with `--`. Now it raises a file not found error.

> Most confusing was the behaviour of `run foo bar` when you have a `run foo baz` running in an other terminal. Because the command exists with code 0 but you do not see anything in your terminal.

ea05b77eea3314cac37c652d5c81090d6fcb7cb5 I added a message of that the instance already running and the arguments that will be launched inside running instance.

> Also the README does not mention that bubblejail does not work with suid bubblewrap.

Do you have a traceback?

**Comment by rusty-snake** [comment]
Running bubblejail with a setuid bwrap shows

```
bwrap: Creating newroot failed: Value too large for defined data type
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib64/python3.11/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/usr/lib64/python3.11/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib64/bubblejail/python_packages/bubblejail/services.py", line 1027, in set_namespaces_limits
    target_namespace = UserNamespace.from_pid(pid)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/bubblejail/python_packages/bubblejail/namespaces.py", line 47, in from_pid
    ns_fd = open_fd(f"/proc/{pid}/ns/user", O_RDONLY | O_CLOEXEC)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/proc/158236/ns/user'
Traceback (most recent call last):
  File "/usr/bin/bubblejail", line 32, in <module>
    bubblejail_main()
  File "/usr/lib64/bubblejail/python_packages/bubblejail/bubblejail_cli.py", line 238, in bubblejail_main
    func(**args_dict)
  File "/usr/lib64/bubblejail/python_packages/bubblejail/bubblejail_cli.py", line 105, in run_bjail
    async_run(
  File "/usr/lib64/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/usr/lib64/bubblejail/python_packages/bubblejail/bubblejail_instance.py", line 252, in async_run_init
    raise BubblewrapRunError((
bubblejail.exceptions.BubblewrapRunError: Bubblewrap failed. Try running bubblejail in terminal to see the exact error.
```

`--dry-run` says
```
--unshare-all --die-with-parent --as-pid-1 --new-session --proc /proc --dev /dev --clearenv --ro-bind /usr /usr --ro-bind /opt /opt --symlink usr/bin /bin --symlink usr/lib /lib --symlink usr/lib64 /lib64 --symlink usr/sbin /sbin --ro-bind /etc /etc --dir /tmp --dir /var --perms 700 --dir /run/user/1000 --bind /home/rusty-snake/.local/share/bubblejail/instances/BubblejailTesting/home /home/rusty-snake --setenv HOME /home/rusty-snake --symlink /home/rusty-snake /home/user --chdir /home/rusty-snake --setenv USER rusty-snake --setenv USERNAME rusty-snake --setenv PATH /usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin --setenv XDG_RUNTIME_DIR /run/user/1000 --setenv LANG de_DE.UTF-8 --setenv DBUS_SESSION_BUS_ADDRESS unix:path=/run/user/1000/bus --bind /run/user/1000/bubblejail/BubblejailTesting/dbus_session_proxy /run/user/1000/bus --seccomp 7 --bind /run/user/1000/bubblejail/BubblejailTesting/dbus_system_proxy /var/run/dbus/system_bus_socket --bind /run/user/1000/bubblejail/BubblejailTesting/dbus_system_proxy /run/dbus/system_bus_socket --info-fd 11
```

and instance config is

```
[namespaces_limits]
user = 0
mount = 0
pid = 0
ipc = 0
net = 0
time = 0
uts = 0
cgroup = 0
```

I've no idea why the `mkdir` fails with `EOVERFLOW`. Actually I was interested of the behaviour of namespace limits. Because bubblewrap creates on one userns if it is suid (if I'm correct).

**Comment by igo95862** [comment]
What about not using `[namespaces_limits]`?

**Comment by rusty-snake** [comment]
> bwrap: Creating newroot failed: Value too large for defined data type

Nevermind, mistake on my side.

**Comment by rusty-snake** [comment]
> What about not using [namespaces_limits]?

I was looking for the behaviour of bubblejail here when it can not apply the restritions. And it just silently ignores them without notice.

**Comment by igo95862** [comment]
> > What about not using [namespaces_limits]?
> 
> I was looking for the behaviour of bubblejail here when it can not apply the restritions. And it just silently ignores them without notice.

Not anymore a371562d4c181317bb1be38b58da97ce29176d08

**Comment by igo95862** [comment]
@rusty-snake can I close the issue?

Technically it is possible for `--debug-shell` in  `bubblejail run BubblejailTesting --debug-shell`  to be treated as bubblejail's option using `nargs="*"` and adding `--` as options terminator.

 However, this will break all already generated desktop entries that have --options and I want to avoid that. In 9dff7653567fe23db5f276a86c4c97247cba1138 I added option separator to all new desktop entries.

To check which desktop entries have `--options` in the `Exec` fields:

```
rg --files-with-matches 'Exec=.*--'  /usr/share/applications/
```

I counted 35 entries on my machine including Firefox and Chromium.
